extensions/VisualEditor/ApiVisualEditor.php

```diff
@@ -63,31 +73,16 @@ private function getVRSObject() {
 		// the VRS class to use, defaults to Parsoid
 		$class = 'ParsoidVirtualRESTService';
 		$config = $this->veConfig;
-		// the global virtual rest service config object, if any
-		$vrs = $this->getConfig()->get( 'VirtualRestConfig' );
-		if ( isset( $vrs['modules'] ) && isset( $vrs['modules']['restbase'] ) ) {
-			// if restbase is available, use it
-			$params = $vrs['modules']['restbase'];
-			$class = 'RestbaseVirtualRESTService';
-			// remove once VE generates restbase paths
-			$params['parsoidCompat'] = true;
-		} elseif ( isset( $vrs['modules'] ) && isset( $vrs['modules']['parsoid'] ) ) {
-			// there's a global parsoid config, use it next
-			$params = $vrs['modules']['parsoid'];
-		} else {
-			// no global modules defined, fall back to old defaults
-			$params = array(
-				'URL' => $config->get( 'VisualEditorParsoidURL' ),
-				'prefix' => $config->get( 'VisualEditorParsoidPrefix' ),
-				'timeout' => $config->get( 'VisualEditorParsoidTimeout' ),
-				'HTTPProxy' => $config->get( 'VisualEditorParsoidHTTPProxy' ),
-				'forwardCookies' => $config->get( 'VisualEditorParsoidForwardCookies' )
-			);
-		}
-		// merge the global and service-specific params
-		if ( isset( $vrs['global'] ) ) {
-			$params = array_merge( $vrs['global'], $params );
-		}
+
+		$params = array(
+			'URL' => $config->get( 'VisualEditorParsoidURL' ),
+			'domain' => $this->getApiSource(),
+			'prefix' => $config->get( 'VisualEditorParsoidPrefix' ),
+			'timeout' => $config->get( 'VisualEditorParsoidTimeout' ),
+			'HTTPProxy' => $config->get( 'VisualEditorParsoidHTTPProxy' ),
+			'forwardCookies' => $config->get( 'VisualEditorParsoidForwardCookies' )
+		);
+
```

Comment:
@pauloslund Would you be able to "minimize" this change - as basically from what I see you want to keep what's in the "else" of original code, and just add "domain" to it. Also - we are most likely to use RESTBase later on (but soon) so that code might be useful again.